src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,363 @@
+     * @return Array of string of size 2 after string is spilt by the delimiter or null if delimiter was not found
+     */
+    private static String[] spiltString(String delimiter, String description) {
+        int byIndex = description.indexOf(delimiter);
+        if (byIndex == -1) {
+            return null;
+        }
+        String taskDescription = description.substring(0, byIndex).trim(); // Remove trailing spaces
+        String date = description.substring(byIndex + delimiter.length()).trim(); // Remove leading spaces
+        String[] output = {taskDescription, date};
+        return output;
+    }
+
+    /**
+     * Adds a Deadline to the Task list
+     *
+     * @param description Task description entered by user.
+     */
+    private static void addDeadLine(String description) {
+        String[] split = spiltString("/by", description);
+        // Handle case it user did not provide date
+        if (split == null) {
+            customPrint("I could not find '/by' in your command!");
+        } else if (split[0].equals("")) {
+            customPrint("I could not find the title of your deadline!");
+        } else {
+            String taskDescription = split[0];
+            String date = split[1];
+            tasks.add(new Deadline(taskDescription, date));
+            acknowledgeCommand();
+        }
+    }
+
+    /**
+     * Adds an Event to the Task list
+     *
+     * @param description Task description entered by user.
+     */
+    private static void addEvent(String description) {
+        String[] split = spiltString("/at", description);
+        // Handle case it user did not provide date
+        if (split == null) {
+            customPrint("I could not find '/at' in your command!");
+        } else if (split[0].equals("")) {
+            customPrint("I could not find the title of your event!");
+        } else {
+            String taskDescription = split[0];
+            String at = split[1];
+            tasks.add(new Event(taskDescription, at));
+            acknowledgeCommand();
+        }
+    }
+
+    /**
+     * Removes a Task from the Task list
+     *
+     * @param id Task id entered by user.
+     */
+    private static void deleteTask(String id) {
+        if (!checkValidTaskId(id)) { // Invalid task id
+            return;
+        }
+
+        int taskId = Integer.parseInt(id) - 1;
+        Task removedTask = tasks.get(taskId);
+        tasks.remove(taskId);
+        customPrint("Noted. I've remove this task:\n" + removedTask.toString() + "\nNow you have "
+                + tasks.size() + " tasks in the list.");
+    }
+
+    /**
+     * Prints the exit message and ends the program.
+     */
+    private static void exit() {
+        customPrint("Bye. Hope to see you again soon!");
+        // Ends the program
+        System.exit(0);
+    }
+
+    /**
+     * Prints the acknowledgement that the command is processed.
+     */
+    private static void acknowledgeCommand() {
```

Comment:
This seems to not acknowledge the command that is processed, but it acknowledges the add command that is processed.